src/Diagnostics.DataProviders/DataProviders/KustoDataProvider.cs

```diff
@@ -140,6 +140,67 @@ public async Task<DataTable> ExecuteQuery(string query, string stampName, string
             var cluster = await GetClusterNameFromStamp(stampName);
             await AddQueryInformationToMetadata(query:query, cluster:cluster, operationName:operationName);
             return await _kustoClient.ExecuteQueryAsync(Helpers.MakeQueryCloudAgnostic(_kustoMap, query), _kustoMap.MapCluster(cluster) ?? cluster, _kustoMap.MapDatabase(_configuration.DBName) ?? _configuration.DBName, requestId, operationName, _queryStartTime, _queryEndTime);
+        }
+
+        private Dictionary<string, Tuple<string, string>> clusterStampNameMapping = new Dictionary<string, Tuple<string, string>>() 
+        {
+            {"wawswus", new Tuple<string, string>("waws-prod-bay-153", "WestUS") },
+            {"wawseus", new Tuple<string, string>("waws-prod-blu-189", "EastUS") },
+            {"wawscus", new Tuple<string, string>("waws-prod-dm1-187", "CentralUS") },
+            {"wawsweu", new Tuple<string, string>("waws-prod-am2-329", "WestEurope") },
+            {"wawsneu", new Tuple<string, string>("waws-prod-db3-169", "NorthEurope") },
+            {"wawseas", new Tuple<string, string>("waws-prod-hk1-029", "EastAsia") }
+        };
+
+        public async Task<DataTable> ExecuteQueryOnFewAppAppServiceClusters(List<string> appServiceClusterNames, string query, string operationName)
+        {
+            if (string.IsNullOrWhiteSpace(operationName))
+            {
+                throw new ArgumentNullException(nameof(operationName), "OperationName cannot be empty. Please supply a name to identify the query.");
+            }
+
+            if (string.IsNullOrWhiteSpace(query))
+            {
+                throw new ArgumentNullException(nameof(query), "Query cannot be empty. Please supply a query to execute.");
+            }
+
+            if (appServiceClusterNames?.Count < 1)
+            {
+                throw new ArgumentNullException(nameof(appServiceClusterNames), "App service kust cluster names list cannot be empty. Please supply at least one kusto cluster name where the query should be executed.");

```

Comment:
nitpik; Spelling

App service **kust** cluster names list cannot be empty. Please supply at least one kusto cluster name where the query should be executed.

Reply:
```suggestion
                throw new ArgumentNullException(nameof(appServiceClusterNames), "App service kusto cluster names list cannot be empty. Please supply at least one kusto cluster name where the query should be executed.");

```